autometa/config/databases.py

```diff
@@ -111,185 +232,388 @@ def format_nr(self):
 
         """
         db_infpath = self.config.get('ncbi', 'nr')
+        db_infpath_md5 = f'{db_infpath}.md5'
         db_outfpath = db_infpath.replace('.gz', '.dmnd')
-        if not self.dryrun and not os.path.exists(db_outfpath):
-            diamond.makedatabase(
-                fasta=nr, database=db_infpath, nproc=self.nproc)
+        checksums_match = False
+        if os.path.exists(db_infpath_md5) and os.path.exists(db_outfpath):
+            # If nr.dmnd.md5 exists, then we will check if nr.gz.md5 is matching
+            current_checksum = read_checksum(db_infpath_md5)
+            current_hash, __ = current_checksum.split()
+            host = self.config.get('ncbi', 'host')
+            remote_checksum = self.get_remote_checksum(host, 'nr')
+            remote_hash, __ = remote_checksum.split()
+            if current_hash == remote_hash:
+                logger.debug(f'nr checksums match, skipping...')
+                checksums_match = True
+        dmnd_md5 = f'{db_outfpath}.md5'
+        do_not_upgrade = os.path.exists(db_outfpath) and not self.upgrade
```

Comment:
The `checksums_match` variable is checked with `do_not_upgrade` in line 250. This way, the user can specify to keep the database _even_ if the checksums _do not_ match, i.e. they do not mind a database that is a few weeks out-of-date. `do_not_upgrade` will only evaluate to True if `nr.dmnd.` exists and the user specifies _not_ to upgrade the database to the most up-to-date version.

Reply:
My point was that if `f'{db_outfpath}.md5'` does not match the current checksum of `db_outfpath` then it should be remade, and perhaps that should be checked here if it would not be caught later?

Reply:
Can you explain the reasoning behind lines 261-266? It looks like if the md5 file exists, then it recalculates the md5 from the diamond file. But then if they don't match the old one is re-written?? Wouldn't this cover up a corrupted file?

Reply:
This I think is the only outstanding issue that has not been resolved yet.